pkg/vsphere/extraconfig/decode.go

```diff
@@ -334,7 +331,7 @@ func decodeMap(src DataSource, dest reflect.Value, prefix string, depth recursio
 	mapkeys, err := src(prefix)
 	if mapkeys == "" || err != nil {
 		log.Debugf("No value found in data source for maps keys \"%s\"", prefix)
-		return this
+		return reflect.ValueOf(nil)
```

Comment:
That precise issue is why I hadn't implemented the nil pointer logic yet. Perhaps we can add a boolean field to decode that denotes whether it's hung from a pointer or not. That would allow us to control on a type by type basis whether you get zero value or nil pointer.